primitives/test-utils/src/keyring.rs

```diff
@@ -0,0 +1,120 @@
+// Copyright 2021 Parity Technologies (UK) Ltd.
+// This file is part of Parity Bridges Common.
+
+// Parity Bridges Common is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity Bridges Common is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity Bridges Common.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Utilities for working with test accounts.
+
+use ed25519_dalek::{Keypair, PublicKey, SecretKey, Signature};
+use finality_grandpa::voter_set::VoterSet;
+use sp_application_crypto::Public;
+use sp_finality_grandpa::{AuthorityId, AuthorityList};
+use sp_runtime::RuntimeDebug;
+
+/// Used to indicate if a type is able to cryptographically sign messages.
+pub trait Signer {
```

Comment:
I don't like that we need to introduce the trait. Can we instead keep just `TestKeyring` and introduce `constants` that represent "named" accounts? Is it really that important to distinguish between `TestKeyring::Alice` and `Account(0)` and introduce additional complexity of handling generic types?